### PR TITLE
[IMP] Improve constrain warning message.

### PR DESCRIPTION
**Pull request by atp-odoo**
Task: https://www.odoo.com/web?#id=21872&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.mSa01Bys9oTToJ5F

### CODE REVIEW

File: openerp/exceptions.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def __init__(self, name, value=None):
             _logger.warn('except_orm is deprecated. Please use specific exceptions like UserError or AccessError. Caller: %s:%s', *caller)
         self.name = name
         self.value = value
+        value = value if value else ''
         self.args = (name, value)
```
Comment (atp-odoo) [inline openerp/exceptions.py]: value = value if value else ''instead of this  pass 
self.args = (name, value or '')

Comment (msh-odoo) [inline openerp/exceptions.py]: @atp-odoo this still under development, checkout task :)

File: addons/website_forum/models/forum.py [hunk 2]
```diff
@@ -134,12 +134,10 @@ def _get_default_faq(self):
...
 
     @api.one
-    @api.constrains('allow_question', 'allow_discussion', 'allow_link', 'default_post_type')
+    @api.constrains('allow_link', 'allow_question', 'allow_discussion', 'default_post_type')
     def _check_default_post_type(self):
-        if (self.default_post_type == 'question' and not self.allow_question) \
-                or (self.default_post_type == 'discussion' and not self.allow_discussion) \
-                or (self.default_post_type == 'link' and not self.allow_link):
-            raise UserError(_('You cannot choose %s as default post since the forum does not allow it.') % self.default_post_type)
+        if self.default_post_type == 'link' and not self.allow_link or self.default_post_type == 'question' and not self.allow_question or self.default_post_type == 'discussion' and not self.allow_discussion:
```
Comment (tivisse) [inline addons/website_forum/models/forum.py]: Is this change really necessary ?

File: addons/website_forum/models/forum.py [hunk 3]
```diff
@@ -134,12 +134,10 @@ def _get_default_faq(self):
...
+            raise ValidationError(_('Post type in "Default post" must be activated'))
```
Comment (tivisse) [inline addons/website_forum/models/forum.py]: Why changing the error message

File: addons/website_forum/models/forum.py [hunk 4]
```diff
@@ -139,7 +139,7 @@ def _check_default_post_type(self):
         if (self.default_post_type == 'question' and not self.allow_question) \
                 or (self.default_post_type == 'discussion' and not self.allow_discussion) \
                 or (self.default_post_type == 'link' and not self.allow_link):
-            raise UserError(_('You cannot choose %s as default post since the forum does not allow it.') % self.default_post_type)
```
Comment (tivisse) [inline addons/website_forum/models/forum.py]: Why removing the value import, this will display "You cannot choose %s as default post since the forum does not allow it."

File: addons/website_forum/models/forum.py [hunk 5]
```diff
@@ -414,7 +414,7 @@ def _check_post_type(self):
         if (self.post_type == 'question' and not self.forum_id.allow_question) \
                 or (self.post_type == 'discussion' and not self.forum_id.allow_discussion) \
                 or (self.post_type == 'link' and not self.forum_id.allow_link):
-            raise UserError(_('This forum does not allow %s') % self.post_type)
+            raise ValidationError(_('This forum does not allow %s' % self.post_type))
```
Comment (tivisse) [inline addons/website_forum/models/forum.py]: I think that it will be difficult to export the translation that way. @mart-e can you confirm ?
